From: Thomas Renninger <trenn@suse.de>
Subject: Blacklist IBM 3850 M2, x3950 M2 to use pci=nocrs
References: bnc#735909
Patch-Mainline: never

Signed-off-by: Thomas Renninger <trenn@suse.de>

Mainline will try to set up PCI resources with use_crs.
But these patches are too new (partly in linux-next,
scheduled for 3.3) and not suitable for SLE11 SP2 (GM) anymore.

Go the safe way and blacklist this specific machine.

---
 arch/x86/pci/acpi.c |   14 ++++++++++++++
 1 file changed, 14 insertions(+)

--- a/arch/x86/pci/acpi.c
+++ b/arch/x86/pci/acpi.c
@@ -24,6 +24,12 @@ static int __init set_use_crs(const stru
 	return 0;
 }
 
+static int __init set_no_crs(const struct dmi_system_id *id)
+{
+	pci_use_crs = false;
+	return 0;
+}
+
 static const struct dmi_system_id pci_use_crs_table[] __initconst = {
 	/* http://bugzilla.kernel.org/show_bug.cgi?id=14183 */
 	{
@@ -72,6 +78,14 @@ static const struct dmi_system_id pci_us
 			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant DL980 G7"),
 		},
 	},
+	{
+		.callback = set_no_crs,
+		.ident = "IBM 3850 M2 / x3950 M2",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "IBM"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "IBM 3850 M2 / x3950 M2"),
+		},
+	},
 	{}
 };
 
